src/components/Icon/FontIcon/FontIcon.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { AriaRole, forwardRef } from "react";
+import classNames from "classnames";
+import VibeComponentProps from "../../../types/VibeComponentProps";
+import VibeComponent from "../../../types/VibeComponent";
+
+interface FontIconProps extends VibeComponentProps {
+  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+  "aria-label"?: string;
+  tabIndex?: number;
+  icon?: string | React.Component | null;
+  role?: AriaRole;
+  "aria-hidden"?: boolean;
+}
+
+const FontIcon: VibeComponent<FontIconProps, HTMLElement> = forwardRef(
+  (
+    {
+      id,
+      className,
+      onClick,
+      "aria-label": iconLabel,
+      tabIndex,
+      icon: Icon,
+      role = "img",
+      "aria-hidden": ariaHidden,
+      "data-testid": dataTestId
+    },
+    iconRef
+  ) => {
+    const isIconFunction = typeof Icon === "function";
+    const iconClassName = isIconFunction ? "" : Icon;
+    return (
+      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+      <span
+        aria-hidden={ariaHidden}
+        className={classNames(className, "fa", iconClassName)}
+        onClick={onClick}
+        ref={iconRef}
+        aria-label={iconLabel}
+        tabIndex={tabIndex}
+        role={role}
+        id={id}
+        data-testid={dataTestId}
+      >
+        {isIconFunction && (
+          //@ts-ignore: Icon components are not converted to typescript in this stage
```

Comment:
I don't think I understand.

`Icon` ultimately comes from the prop `icon`, which is defined as `string` or `React.Component`, this shouldn't have anything to do with components conversion.

What's the original error here?

Reply:
You were right I just used the wrong type, fixed it.
thanks!